flink-java/src/main/java/org/apache/flink/api/java/typeutils/PojoTypeInfo.java

```diff
@@ -310,6 +316,18 @@ public int getFieldIndex(String fieldName) {
 
 	@Override
 	public TypeSerializer<T> createSerializer(ExecutionConfig config) {
+		if(config.isForceKryoEnabled() && config.isForceAvroEnabled()) {
```

Comment:
I don't think that the TypeInfo is the right place to log. This is going to flood your log with 500 identical log messages for larger jobs.